beef_tx_bytes.go

```diff
@@ -78,12 +71,12 @@ func (tx *Transaction) toBeefBytes(compountedPaths CMPSlice) ([]byte, error) {
 	return txBeefBytes, nil
 }
 
-func (tx *Transaction) getCompountedMarklePathIndex(compountedPaths CMPSlice) int {
+func getCompountedMarklePathIndex(tx *bt.Tx, compountedPaths CMPSlice) int {
 	pathIdx := -1
 
 	for i, cmp := range compountedPaths {
 		for txID := range cmp[0] {
-			if txID == tx.ID {
+			if txID == tx.TxID() { // TODO: perf
```

Comment:
On what measures we assume that performance optimisation is needed?

Reply:
In bt, the TxId is calculated every time we try to get it, which means we hash the structure bytes twice each time. It's expensive operation

Reply:
That would be good to include in a comment.

Reply:
I added a more detailed comment